tests/_test_msui/test_mscolab.py

```diff
@@ -188,11 +188,11 @@ def test_add_user(self, mockmessage, qtbot):
 
     @mock.patch("PyQt5.QtWidgets.QMessageBox.question", return_value=QtWidgets.QMessageBox.No)
     def test_add_users_without_updating_credentials_in_config_file(self, mockmessage, qtbot):
+        self._connect_to_mscolab(qtbot)
         create_msui_settings_file('{"MSS_auth": {"' + self.url + '": "something@something.org"}}')
```

Comment:
connect early - more time that it is started

Reply:
The method should be designed in a way that it only returns when the connection is completely working. So how early it is called shouldn't matter, as long as it is before anything that requires the connection. If this indeed changes something then there is a bug and the race condition remains no matter where you move it.

Reply:
see the question with the time.sleep(0.5), I wonder that too.